### PR TITLE
POD-1124 | Restart failed credentials server

### PR DESCRIPTION
**Pull request by janekbaraniewski**
Fixes POD-1124

### CODE REVIEW

File: cmd/ssh.go [hunk 1]
```diff
@@ -647,8 +647,19 @@ func startWorkspaceCredentialServer(ctx context.Context, client *ssh.Client, use
...
 
-	if err := devssh.Run(ctx, client, command, stdin, stdout, writer, nil); err != nil {
-		return fmt.Errorf("run credentials server: %w", err)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Infof("Context canceled, stopping credentials server: %v", ctx.Err())
+			return nil
+		default:
+			if err := devssh.Run(ctx, client, command, stdin, stdout, writer, nil); err != nil {
```
Comment (bkneis) [inline cmd/ssh.go:656]: I think we need a sleep in case of repeated failures

File: pkg/tunnel/services.go [hunk 2]
```diff
@@ -131,10 +132,22 @@ func RunInContainer(
...
-		err = devssh.Run(cancelCtx, containerClient, command, stdinReader, stdoutWriter, writer, nil)
-		if err != nil {
-			return err
+		for {
+			select {
+			case <-cancelCtx.Done():
+				log.Infof("Context canceled, stopping credentials server: %v", cancelCtx.Err())
+				return nil
+			default:
+				err := devssh.Run(cancelCtx, containerClient, command, stdinReader, stdoutWriter, writer, nil)
```
Comment (bkneis) [inline pkg/tunnel/services.go:141]: same here

File: cmd/ssh.go (hunk 1 above)
Comment (janekbaraniewski) [reply cmd/ssh.go:656]: ah a back off mechanism, yeah good point I think we even talked about it. 👀

Comment (pascalbreuninger) [reply cmd/ssh.go:656]: You could use `retry.OnError` from `"k8s.io/client-go/util/retry"` if you want something out of the box

**Review by pascalbreuninger — APPROVED** [review]
LGTM

**Pull request merged**